src/core/device.c

```diff
@@ -735,37 +741,112 @@ static bool device_is_ready(sd_device *dev) {
         return r != 0;
 }
 
-static int device_setup_devlink_unit_one(Manager *m, const char *devlink, sd_device **ret) {
+static int device_setup_alias_units(Manager *m, sd_device *dev, Set **ready_units, Set **not_ready_units) {
+        _cleanup_strv_free_ char **aliases = NULL;
+        const char *syspath;
+        Device *l;
+        int r;
+
+        assert(m);
+        assert(dev);
+        assert(ready_units);
+        assert(not_ready_units);
+
+        /* On remove, all alias units will be removed by device_update_found_by_sysfs() in device_dispatch_io(). */
+        if (device_for_action(dev, SD_DEVICE_REMOVE))
+                return 0;
+
+        r = sd_device_get_syspath(dev, &syspath);
+        if (r < 0)
+                return r;
+
+        if (device_is_ready(dev)) {
+                const char *s;
+
+                r = sd_device_get_property_value(dev, "SYSTEMD_ALIAS", &s);
+                if (r < 0 && r != -ENOENT)
+                        log_device_warning_errno(dev, r, "Failed to get SYSTEMD_ALIAS property, ignoring: %m");
+                if (r >= 0) {
+                        r = strv_split_full(&aliases, s, NULL, EXTRACT_UNQUOTE);
+                        if (r < 0)
+                                log_device_warning_errno(dev, r, "Failed to parse SYSTEMD_ALIAS property, ignoring: %m");
+                }
+        }
+
+        STRV_FOREACH(alias, aliases) {
+                if (!path_is_absolute(*alias)) {
+                        log_device_warning(dev, "SYSTEMD_ALIAS is not an absolute path, ignoring: %s", *alias);
```

Comment:
This was already the case, but the warning message here seems wrong, this is not about SYSTEMD_ALIAS in its entirety, but a single alias in it.